src/lib/utils/padScale.js

```diff
@@ -1,21 +1,37 @@
+import isOrdinalDomain from '../helpers/isOrdinalDomain.js';
+import getPadFunctions from '../helpers/getPadFunctions.js';
+import findScaleName from '../helpers/findScaleName.js';
+
+// These scales have a discrete range so they can't be padded
+const unpaddable = ['scaleThreshold', 'scaleQuantile', 'scaleQuantize', 'scaleSequentialQuantile'];
+
+/**
+ * @typedef {import('d3-scale').ScaleLinear<any, any> |
```

Comment:
Here I went through the d3-docs and added the interface definitions.

Not sure whether one should exclude those that are filtered out with `unpaddable` or whether there is already combinations like `scaleContinuousNumeric` which would make the list shorter.

Reply:
I think, even though the script will handle an unpaddable scale, that it should be a typescript error if you try to pass in an unpaddable scale. 

I didn't follow the second part of what you were saying about `scaleContinuous Numeric`. Can you explain what you mean more?

Reply:
Sorry I should have linked that. in d3-typed there is [scaleContinousNumeric](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/5d86c5afed68b6dfdaa07fcec9bf9b99325410a2/types/d3-scale/index.d.ts#L32), which then gets extended to `ScaleLinear`, `ScalePower`, `ScaleLogarithmic` and a few more.